### PR TITLE
[Integration Tests] Fix GenerateBlockManually

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin.IntegrationTests/SmartContracts/SmartContractWalletTests.cs [hunk 1]
```diff
@@ -89,7 +88,8 @@ public void SendAndReceiveCorrectly()
                 Assert.Null(scReceiver.FullNode.WalletManager().GetSpendableTransactionsInWallet(WalletName).First().Transaction.BlockHeight);
 
                 // generate two new blocks do the trx is confirmed
-                scSender.GenerateBlockManually(new List<Transaction>(new[] { scSender.FullNode.Network.CreateTransaction(trx.ToBytes()) }));
+                scSender.AddToStratisMempool(scSender.FullNode.Network.CreateTransaction(trx.ToBytes()));
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/SmartContracts/SmartContractWalletTests.cs]: Is this to follow up on what we talked about in the previous PR? (add to mempool then mine)

Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/SmartContracts/SmartContractWalletTests.cs]: Hi Jeremy, yes that is correct 👍

Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests/SmartContracts/SmartContractWalletTests.cs]: I was able to change some of the tests to use this.

File: src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs [hunk 2]
```diff
@@ -311,7 +311,6 @@ public void MiningNodeWithOneConnectionAlwaysSynced()
                 Thread.Sleep(1);
 
             // Make sure that miner did not advance yet but connector did.
-            Assert.NotEqual(miner.FullNode.Chain.Tip.HashBlock, networkNode1.FullNode.Chain.Tip.HashBlock);
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs]: I agree that these look a bit flaky:
- go fast and the miner will be equal to networkNode1
- go slow and it won't be equal

At the same time, a slow or fast computer will mean slow/fast propagation but also slow/fast pass through the assertions.

I think this needs to be redesigned to either wait for nodes to be in sync or only connect nodes after one of them has moved forward.

**Pull request merged**